libcloud/compute/drivers/vsphere.py

```diff
@@ -49,506 +36,228 @@
 
-        :return: True on success.
-        :rtype: ``bool``
-        """
-        vm = self._get_vm_for_node(node=node)
-        vm.migrate(priority=priority, resource_pool=resource_pool, host=host)
+        if ip_address:
+            try:
+                if is_public_subnet(ip_address):
+                    public_ips.append(ip_address)
+                else:
+                    private_ips.append(ip_address)
+            except:
+                # IPV6 not supported
+                pass
 
-        return True
+        node = Node(id=uuid, name=name, state=status,
+                    public_ips=public_ips, private_ips=private_ips,
+                    driver=self, extra=extra)
+        node._uuid = uuid
+        return node
 
-    @wrap_non_libcloud_exceptions
     def reboot_node(self, node):
-        vm = self._get_vm_for_node(node=node)
-        vm.reset()
-
-        return True
-
-    @wrap_non_libcloud_exceptions
-    def destroy_node(self, node, ex_remove_files=True):
         """
-        :param ex_remove_files: Remove all the files from the datastore.
-        :type ex_remove_files: ``bool``
         """
-        ex_remove_files = False
-        vm = self._get_vm_for_node(node=node)
-
-        server = self.connection.client
-
-        # Based on code from
-        # https://pypi.python.org/pypi/pyxenter
-        if ex_remove_files:
-            request = VI.Destroy_TaskRequestMsg()
-
-            _this = request.new__this(vm._mor)
-            _this.set_attribute_type(vm._mor.get_attribute_type())
-            request.set_element__this(_this)
-            ret = server._proxy.Destroy_Task(request)._returnval
-            task = VITask(ret, server)
-
-            # Wait for the task to finish
-            status = task.wait_for_state([task.STATE_SUCCESS,
-                                          task.STATE_ERROR])
-
-            if status == task.STATE_ERROR:
-                raise LibcloudError('Error destroying node: %s' %
-                                    (task.get_error_message()))
-        else:
-            request = VI.UnregisterVMRequestMsg()
-
-            _this = request.new__this(vm._mor)
-            _this.set_attribute_type(vm._mor.get_attribute_type())
-            request.set_element__this(_this)
-            ret = server._proxy.UnregisterVM(request)
-            task = VITask(ret, server)
-
-        return True
-
-    @wrap_non_libcloud_exceptions
-    def ex_stop_node(self, node):
-        vm = self._get_vm_for_node(node=node)
-        vm.power_off()
-
-        return True
-
-    @wrap_non_libcloud_exceptions
-    def ex_start_node(self, node):
-        vm = self._get_vm_for_node(node=node)
-        vm.power_on()
-
-        return True
-
-    @wrap_non_libcloud_exceptions
-    def ex_suspend_node(self, node):
-        vm = self._get_vm_for_node(node=node)
-        vm.suspend()
-
+        vm = self.find_by_uuid(node)
+        try:
+            vm.RebootGuest()
+        except:
+            pass
```

Comment:
can you catch a specific error type here and either rethrow an error or return false.